### ISSUE TITLE
may I ask a question about the algorithm ?

### ISSUE THREAD

**Issue by 1025616969** [cbfinn/maml#36]
hello ,big gun, I'm new in this area, after read your paper MAML,I am confused why you update theta in the inner loop to get theta' , and then in outer loop, you use theta' to calculate final loss.
In my opinion, if i consider all the tasks as a batch , and use original theta to calculate each task 's loss,and then sum up to do gradient descent, in this way it's also reasonable to find a proper theta that suits all the tasks,because it makse the sum of all the tasks loss minimal,
I want to know the insight your algorithm do and the advantages of your algorithm compared to my opinoin.
Thanks a lot ,  maybe problem seems stupid ,looking forward to your reply, hahaha!

**Comment by cbfinn** [comment]
This comparison is provided in the paper for some of the experimental domains, and does not perform well in settings where the tasks are mutually exclusive, as you would expect. All of the experimental settings in this paper study task distributions with mutually exclusive tasks. In settings where the tasks are not mutually exclusive, this comparison is expected to perform strongly.

**Comment by 1025616969** [comment]
Thanks for replying,really convincing.
You update the parameters using meta-train, and use this updated parameters to get loss on meta-test, and sum up.When tasks are mutually exclusive, why this policy works?when tasks are exclusive,then the losses you calculated on meta-test will be exclusive ,too. Right? So why your policy can work so good?So i still dont know the insight difference between your policy and my opinion.dose make a gradient descent in advance make any sense?

**Comment by 1025616969** [comment]
do anyone give some explanations? really confused，yayayayayay

**Comment by zwd973** [comment]
because he want to find the theta including the commom distribution from mutually exclusive tasks.
